### ISSUE TITLE
Sleep function not work for dummy table

### ISSUE THREAD

**Issue by sundy-li** [datafuselabs/datafuse#974]
**Summary**

Sleep function not work for dummy table

**How to reproduce**
 

```

datafuse :) select count(sleep(2))  from numbers(10);

SELECT count(sleep(2))
FROM numbers(10)

Query id: 978eada9-86ab-4dbb-b947-a8ee2ffa3b3f

┌─count(sleep(2))─┐
│              10 │
└─────────────────┘

1 rows in set. Elapsed: 2.112 sec.

datafuse :) select sleep(333);

SELECT sleep(333)

Query id: f66c9576-d8d9-4447-b078-b5cae8ba2f9a

Ok.

0 rows in set. Elapsed: 0.125 sec.

datafuse :)
```

**Comment by sundy-li** [comment]
cc @zhang2014

**Comment by PsiACE** [comment]
I use the mysql client to connect to datafuse and cannot reproduce this problem.

> The maximum sleep time is 3 seconds. Requested: 3s.

**Comment by sundy-li** [comment]
So it maybe the bug of clickhouse_handler.

**Comment by BohuTANG** [comment]
The clickhouse handler has some issues with the error/panic handling, maybe we can address them all in another patch.
cc @sundy-li @zhang2014

**Comment by zhang2014** [comment]
They should be fixed in #931
